app/src/features/Batches/BatchResourceDialog.tsx

```diff
@@ -0,0 +1,238 @@
+import React, { useState, useEffect, useMemo } from "react";
+
+import {
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  InputAdornment,
+  List,
+  makeStyles,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import { Search } from "@material-ui/icons";
+import clsx from "clsx";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router";
+
+import Button from "common/components/Button";
+import {
+  useApiBatchesCreateMutation,
+  useApiResourcesListQuery,
+} from "services/api/endpoints";
+
+import BatchResourceListItem from "./BatchResourceListItem";
+
+const useStyles = makeStyles((theme) => ({
+  dialog: {
+    padding: theme.spacing(3),
+    height: 470,
+  },
+  title: {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+    flexWrap: "wrap",
+  },
+  checkboxForm: {
+    cursor: "pointer",
+  },
+  textField: {
+    padding: theme.spacing(0, 3),
+  },
+}));
+
+type BatchResourceDialogType = {
+  open: boolean;
+  handleClose: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
When setting functions as component props, best practice. would be to prefix them with `on` instead of `handle`.

More over, you can be less specific about its type, because nothing forbids us to set something different from a React.Dispatch function :

```suggestion
  onClose: (boolean) => void;
```